File: solutions/haskell/097-number-of-1-bits.hs
{-
  Problem 97: Number of 1 Bits (LeetCode 191)
  Difficulty: Easy
  Language: Haskell
-}
import Data.Bits

hammingWeight :: Int -> Int
hammingWeight 0 = 0
hammingWeight n = (n .&. 1) + hammingWeight (shiftR n 1)

File: solutions/rust/097-number-of-1-bits.rs
/**
 * Problem 97: Number of 1 Bits (LeetCode 191)
 * Difficulty: Easy
 * Language: Rust
 */
fn hamming_weight(n: u32) -> u32 {
    if n == 0 { 0 } else { (n & 1) + hamming_weight(n >> 1) }
}
